test/e2e/framework/statefulset/rest.go

```diff
@@ -268,6 +269,54 @@ func CheckHostname(c clientset.Interface, ss *appsv1.StatefulSet) error {
 	return nil
 }
 
+// CheckIdentityLabels verifies that all Pods in ss have correct and non-overalapping
+// identity labels. If the returned error is not nil then verification failed.
+func CheckIdentityLabels(c clientset.Interface, ss *appsv1.StatefulSet) error {
+	podList := GetPodList(c, ss)
+	knownLabelValues := make(map[string]bool, len(podList.Items)*2)
+
+	for _, statefulPod := range podList.Items {
+		// Verify pod name label
+		podNameLabel, ok := statefulPod.Labels[appsv1.StatefulSetPodNameLabel]
+		if !ok {
+			return fmt.Errorf("missing %s label for pod %s", appsv1.StatefulSetPodNameLabel, statefulPod.Name)
+		}
+		if podNameLabel != statefulPod.Name {
+			return fmt.Errorf(
+				"expected %s label value (%s) and stateful pod name (%s) to be equal",
+				appsv1.StatefulSetPodNameLabel, podNameLabel, statefulPod.Name,
+			)
+		}
+		if knownLabelValues[podNameLabel] {
```

Comment:
Is this check necessary (considering the name check on line 284) ?

Reply:
looks right as this is the non overlapping check